contracts/BC_fusion/StakeCredit.sol

```diff
@@ -0,0 +1,228 @@
+        if (totalPooledBNB == 0) {
+            return 0;
+        }
+        return (bnbAmount * totalSupply()) / totalPooledBNB;
+    }
+
+    /**
+     * @return the amount of BNB that corresponds to `_sharesAmount` token shares.
+     */
+    function getPooledBNBByShares(uint256 shares) public view returns (uint256) {
+        if (totalSupply() == 0) {
+            return 0;
+        }
+        return (shares * totalPooledBNB) / totalSupply();
+    }
+
+    function unbondRequest(address delegator, uint256 _index) public view returns (UnbondRequest memory, uint256) {
+        bytes32 hash = _unbondRequestsQueue[delegator].at(_index);
+        return (_unbondRequests[hash], _unbondRequestsQueue[delegator].length());
+    }
+
+    function lockedBNBs(address delegator) public view returns (uint256) {
+        uint256 length = _unbondRequestsQueue[delegator].length();
+        if (length == 0) {
+            return 0;
+        }
+
+        uint256 _totalBnbAmount;
+        for (uint256 i; i < length; ++i) {
+            bytes32 hash = _unbondRequestsQueue[delegator].front();
+            UnbondRequest memory request = _unbondRequests[hash];
+            _totalBnbAmount += request.bnbAmount;
+        }
+        return _totalBnbAmount;
+    }
+
+    function unbondSequence(address delegator) public view returns (uint256) {
+        return _unbondSequence[delegator].current();
+    }
+
+    function getSelfDelegationBNB() public view returns (uint256) {
+        return getPooledBNBByShares(balanceOf(validator));
+    }
+
+    function getPooledBNB(address account) external view returns (uint256) {
+        return getPooledBNBByShares(balanceOf(account));
+    }
+
+    /*----------------- internal functions -----------------*/
+    function _bootstrapInitialHolder(uint256 initAmount) internal onlyInitializing {
+        require(validator != address(0), "INVALID_VALIDATOR");
+        require(totalSupply() == 0, "TOTAL_SUPPLY_NOT_ZERO");
+
+        // mint initial tokens to the validator
+        // shares is equal to the amount of BNB staked
+        _mint(validator, initAmount);
+        totalPooledBNB = initAmount;
+    }
+
+    function _mintAndSync(address account, uint256 bnbAmount) internal returns (uint256 shares) {
+        shares = getSharesByPooledBNB(bnbAmount);
+        _mint(account, shares);
+        totalPooledBNB += bnbAmount;
+    }
+
+    function _burnAndSync(address account, uint256 shares) internal returns (uint256 bnbAmount) {
+        bnbAmount = getPooledBNBByShares(shares);
+        _burn(account, shares);
+        totalPooledBNB -= bnbAmount;
+    }
+
+    function _useSequence(address delegator) internal returns (uint256 current) {
+        CountersUpgradeable.Counter storage sequence = _unbondSequence[delegator];
+        current = sequence.current();
+        sequence.increment();
+    }
+
+    function _transfer(address, address, uint256) internal pure override {
```

Comment:
can you write a testcase to ensure this take effect, because there is no "override" key word here, I am not sure.

Reply:
There's `override`.